daemon/cluster/cluster.go

```diff
@@ -893,6 +893,17 @@ func (c *Cluster) UpdateService(serviceID string, version uint64, spec types.Ser
 		return err
 	}
 
+	// Check if anything is modified in spec for update #25770
+	currentService, err := getService(ctx, c.client, serviceID)
+	if err != nil {
+		return err
+	}
+
+	if serviceSpec.String() == currentService.String() {
```

Comment:
@AkihiroSuda Agree Strings in general not optimal for comparing. But ``serviceSpec.String()`` https://github.com/docker/swarmkit/blob/master/api/specs.pb.go#L2059 is swarmkit Function , so used.

I tried using reflect.DeepEqual() which was failing(coudn't verify why), so used this.